pwnlib/elf/pltgot.py

```diff
@@ -0,0 +1,291 @@
+"""Utilities to parse the PLT and GOT of an ELF file.
+
+Style note: these utilities should work with plain elftools ELFFiles.
+"""
+
+from elftools.elf.constants import SHN_INDICES
+from elftools.elf.relocation import RelocationSection
+from pwnlib.util import packing
+from pwnlib.log import getLogger
+
+log = getLogger(__name__)
+
+def read_got(elf):
+    """ Read GOT symbols and corresponding program locations from the binary. """
+
+    got = {}
+
+    # TODO: In reality, the canonical relocation list is the one linked from
```

Comment:
I was confusing `.dynamic` (the section) with `_DYNAMIC` (the symbol at the beginning of the GOT)